search-improvements.rst

```diff
@@ -0,0 +1,189 @@
+
+- `Search as You Type and Autocomplete`_
+- `Code Search`_ 
+
+All of my search related work can be seen in the `Search Project Board`_.
+
+
+Background
+^^^^^^^^^^
+Search is a vital part of any documentation hosting platform, so people can get the
+information they need. As a documentation hosting platform, the same rules apply to
+Read The Docs. Because of having a small core team, the search functionality
+of Read The Docs has lagged behind for quite a while now. Initially the search code
+was voluntarily contributed by `Rob Hudson`_,  `back in 2013`_ and then improved by other
+contributors. The search infrastructure was already outdated as
+Read The Docs were using `Elasticsearch 1.3.x`_ which was already reached its
+End of Life in 2016. Therefore, the upgradation of search infrastructure was badly needed.
+
+Built in Search vs Read The Docs Search
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Both Sphinx and MkDocs already have built in search functionality. But the features are very limited.
+At Read The Docs, we have felt the limitations and therefore we index the documentations in our
+Elasticsearch_ index so that we can provide better search experience like:
+
+- Search across multiple projects
+- Advanced query syntax
+- Search inside subprojects
+- Improved search result order
+- Public Search API (Documentation pending)
+
+**Currently, we do not index MkDocs documents with** Elasticsearch_,
+but `any kind of help is welcome`_.
+
+New Features
+^^^^^^^^^^^^
+In the 3 months of full time work, I have implemented couple of features including
+many bug fixes. Some of the major features are as following:
+
+- `Exact Matching Search`_: Exact matching is one of our most highly requested
+  features for Read the Docs. Now you can search for exact word in documentations
+  by having your query inside quotation (`""`). So if you search
+  for `"Here is foo"` (with the quotation), you will get all the documentations where the full
+  `Here is foo` phrase exist.
+
+- Simple Query String Syntax support: Now you can use `Simple Query String Syntax`_ for
+  searching. For example, if you search with `Mozilla +-Firefox`, you will get documentations
+  where the `Mozilla` word is present, but `Firefox` word is not present.
+  For more information, you can look at `Simple Query String Syntax`_ documentation.
+
+- `Case Insensitive Search`_: The search is now case insensitive. So if you search for `Foo`,
+  you will get all the documentations which have one of either `Foo`, `FOO`, `foo` or `fOO`.
+
+- `Improved Result order`_: The result order is improved dramatically. Therefore if you Search
+  with `Foo Bar`, you will first see the documentations which have both `Foo Bar`, then
+  you will see other documentations which have either `Foo` or `Bar`
+
+- `Auto Removing from Search Index`_: In the past, if any page got removed from documentations,
+  the page was still available in documentations. But from now on, the page will be removed
```

Comment:
documentations -> documentation

Reply:
If you want to mix up your verbiage a bit, you can use "docs" but "documentations" (while technically a word) is extremely rare and probably shouldn't be used.